### PR TITLE
Limit deprovisioned instances

### PR DESCRIPTION
**Pull request by KsaweryZietara**
**Description**

Changes proposed in this pull request:

- revert changes from https://github.com/kyma-project/kyma-environment-broker/pull/1022,
- limit deprovisioned instances in client.

**Related issue(s)**
See also https://github.com/kyma-project/kyma-environment-broker/issues/1021

### CODE REVIEW

File: common/runtime/client.go [hunk 1]
```diff
@@ -12,7 +12,10 @@ import (
 	"github.com/kyma-project/kyma-environment-broker/common/pagination"
 )
 
-const defaultPageSize = 100
+const (
+	defaultPageSize = 100
+	RuntimesLimit   = 1000
```
Comment (jaroslaw-pieszka) [inline common/runtime/client.go:17]: So we leave it hard-wired, no configuration possible?

File: common/runtime/client.go [hunk 2]
```diff
@@ -91,6 +102,12 @@ func (c *client) ListRuntimes(params ListParameters) (RuntimesPage, error) {
 		} else {
 			fetchedAll = true
 		}
+
+		if limitRuntimes && runtimes.Count >= RuntimesLimit {
+			runtimes.Data = runtimes.Data[:RuntimesLimit]
+			runtimes.Count = RuntimesLimit
+			fetchedAll = true
```
Comment (jaroslaw-pieszka) [inline common/runtime/client.go:109]: Is this meant just to force break out of the for loop?

File: common/runtime/client_test.go [hunk 3]
```diff
@@ -118,6 +118,32 @@ func TestClient_ListRuntimes(t *testing.T) {
 		assert.Equal(t, 4, rp.TotalCount)
 		assert.Len(t, rp.Data, 4)
 	})
+
+	t.Run("Test deprovisioned runtimes limit", func(t *testing.T) {
+		//given
+		params := ListParameters{
```
Comment (jaroslaw-pieszka) [inline common/runtime/client_test.go:124]: Do we test an execution path where there is no `StateDeprovisioned` in the params, so the limit has no effect (`limitRuntimes` is false).

File: common/runtime/client.go (hunk 1 above)
Comment (KsaweryZietara) [reply common/runtime/client.go:17]: I think this parameter will not change in the future, so I would leave it hard-wired. Do you think it should be configurable?

File: common/runtime/client_test.go (hunk 3 above)
Comment (KsaweryZietara) [reply common/runtime/client_test.go:124]: There are two tests without `StateDeprovisioned` in the params: 
- https://github.com/kyma-project/kyma-environment-broker/blob/main/common/runtime/client_test.go#L35
- https://github.com/kyma-project/kyma-environment-broker/blob/main/common/runtime/client_test.go#L97

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**